Fix tags in generic order not expanded

HO-COOH opened the pull request:
# Fix/Feature/Other

## Fix

- [ ] Link to issue.
This closes #204 

- [ ] Description of the fix
~~Copied file-scoped tag branches -> generic tag branches, so it can be properly expanded.~~
The common tags that can be used in both `File` comments and `Method` comments are handled by one same function. So it can be both expanded properly.
![genericTags](https://user-images.githubusercontent.com/42881734/116768652-d8791000-aa6a-11eb-9021-1eca9fd2e95c.gif)

cschlosser reviewed (changes requested):
I would prefer if we found a solution where we can reuse the already existing `generateCustomTag` method.

Otherwise the settings will be confusing and this breaks current functionality that people may rely on.

Maybe we have to make `generateCustomTag` smarter to replace what's possible but also insert tag's it doesn't have a replacement for.

HO-COOH commented:
Cleaned up a little, and as you said `generateCustomTag()` is now reused.
I also moved those common tags that shared between `File` and `Method` to a separate function so easier to maintain, *hopefully*.

cschlosser commented on src/Lang/Cpp/CppDocGen.ts:
```diff
@@ -502,13 +508,11 @@ export class CppDocGen implements IDocGen {
                 }
                 default: {
-                    break;
+                    this.generateCommonTag(lines, element);
```
Can we let the default case still do nothing?

cschlosser commented on src/Lang/Cpp/CppDocGen.ts:
```diff
@@ -408,40 +446,16 @@ export class CppDocGen implements IDocGen {
                 }
                 default: {
-                    break;
+                    this.generateCommonTag(lines, element);
```
I would prefer leaving the default case empty. Otherwise typo's may show up with the custom tags which is probably not what de user expects.

HO-COOH replied:
That's handled in `generateCommonTag()`. There is a default break case there.

HO-COOH replied:
That's handled in `generateCommonTag()`. There is a default break case there.

cschlosser replied:
My bad. I misread `generateCommonTag` as `generateCustomTag`.

cschlosser reviewed (commented):
Thank you for extracting the shared one's. That's good I think.
Could you please take a look at CI so we can merge it?

HO-COOH commented:
I see the error in CI comes from tslint. You might consider migrating to eslint. I am not very familiar with these toolings, just a suggestion :)

cschlosser reviewed (commented):
Looks good.
One more thing though. Could you please write a reproducer test for the bug?

You wrote here https://github.com/cschlosser/doxdocgen/issues/204#issuecomment-830499301 that `doxdocgen.generic.customTags` have to be removed from the config. This should no longer be true. Can we have a test for that as well please?

HO-COOH commented:
> 
> 
> Looks good.
> One more thing though. Could you please write a reproducer test for the bug?
> 
> You wrote here [#204 (comment)](https://github.com/cschlosser/doxdocgen/issues/204#issuecomment-830499301) that `doxdocgen.generic.customTags` have to be removed from the config. This should no longer be true. Can we have a test for that as well please?

It's still true. He has duplicate tags in `generic.order` and `generic.customTags` such as `author` and `date` in the case. It's fine tho, if he really wants to include them twice. 

The change for this issue is that those variables in `generic.customTags` like `{date}` would be expanded in this PR.

HO-COOH commented:
Added test case for this. And found and fixed another bug, that is if gitConfig is for user name and email is used, the custom tag with `{author}` and `{email}` will not be expanded properly, because your code is using `this.cfg.authorName` in the `getMultiTemplateString()`
